hardware_pydantic/devices.py

```diff
@@ -64,26 +114,114 @@ def post__transfer(
             self,
             from_obj: Rack | Heater,
             to_obj: Rack | Heater,
-            transferee: Vial,
+            vial: Vial,
             to_position: str | None
     ) -> None:
+        """Transfer a vial from one place to another."""
         # TODO make more general
         # take it out
-        assert transferee.position_relative == from_obj.identifier
+        assert vial.position_relative == from_obj.identifier
         if isinstance(from_obj, Rack):
-            assert transferee.position in from_obj.content
-            from_obj.content[transferee.position] = None
+            assert vial.position in from_obj.content
+            from_obj.content[vial.position] = None
         elif isinstance(from_obj, Heater):
             from_obj.content = None
         else:
             raise TypeError
-        transferee.position = None
-        transferee.position_relative = None
+        vial.position = None
+        vial.position_relative = None
         # put it to
         if isinstance(to_obj, Rack):
             assert to_position in to_obj.content
-            to_obj.content[to_position] = transferee.identifier
+            to_obj.content[to_position] = vial.identifier
         elif isinstance(to_obj, Heater):
-            to_obj.content = transferee.identifier
-        transferee.position = to_position
-        transferee.position_relative = to_obj.identifier
+            to_obj.content = vial.identifier
+        vial.position = to_position
+        vial.position_relative = to_obj.identifier
+
+
+class PlateGripperArm(Device):
+    pass
+
+
+class SolidSVTool(Device):
+    """Solid sample vial tool for transferring solids."""
+    # todo: this device is often used together with a balance
+    # note: this is not used for the current simulation
+    def pre__transfer(self, from_obj: Vial, to_obj: Vial, amount: float):
+        """Transfer solids from one vial to another."""
+        # TODO sample from a dist
+        removed = from_obj.remove_content(amount)
+        to_obj.add_content(removed)
+
+    def post__transfer(self, from_obj: Vial, to_obj: Vial, amount: float):
+        """Time needed to transfer solids from one vial to another."""
+        # todo: add adaptive post to estimate the time of transferring solids
+        # transfer speed in g/s
+        transfer_speed = 2
+        return amount / transfer_speed
+
+
+class Balance(Device):
```

Comment:
on Junior balance is just a part of solid dispense. you can't even access its reading at runtime and tare is done automatically. a balance should be just considered a rack slot.